source/jormungandr/jormungandr/scenarios/utils.py

```diff
@@ -461,3 +463,91 @@ def include_poi_access_points(request, pt_object, mode):
         ]
         and pt_object.poi.children
     )
+
+
+def get_impact_uris_for_poi(response, poi):
+    impact_uris = set()
+    if response is None:
+        return impact_uris
+    for impact in response.impacts:
+        for object in impact.impacted_objects:
+            if object.pt_object.embedded_type == type_pb2.POI and object.pt_object.uri == poi.uri:
+                impact_uris.add(impact.uri)
+                object.pt_object.poi.CopyFrom(poi)
+
+    return impact_uris
+
+
+def fill_disruptions_on_pois(instance, response):
+    if not response.pois:
+        return
+
+    # add all poi_ids as parameters
+    poi_uris = set()
+    for poi in response.pois:
+        poi_uris.add(poi.uri)
+
+    # calling loki with api poi_disruptions
+    resp_poi = get_disruptions_on_poi(instance, poi_uris)
+
+    # For each poi in the response add impact_uris from resp_poi
+    # and copy object poi in impact.impacted_objects
+    for poi in response.pois:
+        impact_uris = get_impact_uris_for_poi(resp_poi, poi)
+        for impact_uri in impact_uris:
+            poi.impact_uris.append(impact_uri)
+
+    # Add all impacts from resp_poi to the response
+    add_disruptions(response, resp_poi)
+
+
+def fill_disruptions_on_places_nearby(instance, response):
+    if not response.places_nearby:
+        return
+
+    # Add all the poi uris in a list
+    poi_uris = set()
+    for place_nearby in response.places_nearby:
+        if place_nearby.embedded_type == type_pb2.POI:
+            poi_uris.add(place_nearby.uri)
+
+    # calling loki with api poi_disruptions
+    resp_poi = get_disruptions_on_poi(instance, poi_uris)
+
+    # For each poi in the response add impact_uris from resp_poi
+    # and copy object poi in impact.impacted_objects
+    for place_nearby in response.places_nearby:
+        if place_nearby.embedded_type == type_pb2.POI:
+            impact_uris = get_impact_uris_for_poi(resp_poi, place_nearby.poi)
+            for impact_uri in impact_uris:
+                place_nearby.poi.impact_uris.append(impact_uri)
+
+    # Add all impacts from resp_poi to the response
+    add_disruptions(response, resp_poi)
+
+
+def get_disruptions_on_poi(instance, uris, since_datetime=None, until_datetime=None):
+    if not uris:
```

Comment:
what if both since_datetime and until_datetime are None?

Reply:
We don't initialize parameters if they are None.